.github/workflows/release.yaml

```diff
@@ -25,7 +26,7 @@ jobs:
         id: cache
         with:
           path: ${{ env.pythonLocation }}
-          key: ${{ env.pythonLocation }}-${{ hashFiles('pyproject.toml') }}-integration
+          key: ${{ runner.os }}-${{ runner.arch }}-Python-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('pyproject.toml') }}
```

Comment:
I would love to add OS version too, but I haven't yet figured out where to find that.

We need the full X.Y.Z Python version, X.Y is not sufficient.